### PR TITLE
updated multi-tenancy example to add TenantRepository and corrected t…

### PR DESCRIPTION
**Pull request by alexandre-melard**
Corrected some typos in the documentation and added an example TenantRepository

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@mylen Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=7643)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security&pullRequestId=7643) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@mylen Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=7643)!

File: docs/manual/src/docs/asciidoc/_includes/servlet/oauth2/oauth2-resourceserver.adoc [hunk 1]
```diff
@@ -1273,13 +1296,13 @@ public class TenantAuthenticationManagerResolver implements AuthenticationManage
 
 	@Override
 	public AuthenticationManager resolve(HttpServletRequest request) {
-		return this.authenticationManagers.computeIfAbsent(toTenant(request), this::fromTenant); <3>
```
Comment (jzheaux) [inline docs/manual/src/docs/asciidoc/_includes/servlet/oauth2/oauth2-resourceserver.adoc]: Why was this removed? It should be linked to the `<3>` in the comments below.

File: docs/manual/src/docs/asciidoc/_includes/servlet/oauth2/oauth2-resourceserver.adoc [hunk 2]
```diff
@@ -1258,6 +1258,29 @@ http
 Resolving the tenant by claim is similar to doing so by request material.
 The only real difference is the `toTenant` method implementation:
 
+[source,java]
+----
+@Component
+public class TenantRepository {
```
Comment (jzheaux) [inline docs/manual/src/docs/asciidoc/_includes/servlet/oauth2/oauth2-resourceserver.adoc]: I see your point in adding this sample `TenantRepository`. But, I think this demonstrates actually that it would be better to change the classes below to simply use a pair of `Map`s.

Would you mind changing the examples to use a `Map<String, String>` instead? Then, the members might be called `issuersByTenant` and `jwkUrisByTenant` instead of `tenants` or `tenantRepository`.

File: docs/manual/src/docs/asciidoc/_includes/servlet/oauth2/oauth2-resourceserver.adoc [hunk 3]
```diff
@@ -1258,6 +1258,29 @@ http
 Resolving the tenant by claim is similar to doing so by request material.
 The only real difference is the `toTenant` method implementation:
 
+[source,java]
```
Comment (jzheaux) [inline docs/manual/src/docs/asciidoc/_includes/servlet/oauth2/oauth2-resourceserver.adoc]: I think this snippet might be in the wrong place. Note the previous sentence, which is introducing the code snippet that is now below this one. I think we'd need to change the description a bit, too, for the reader to be able to follow.

File: docs/manual/src/docs/asciidoc/_includes/servlet/oauth2/oauth2-resourceserver.adoc (hunk 1 above)
Comment (alexandre-melard) [reply docs/manual/src/docs/asciidoc/_includes/servlet/oauth2/oauth2-resourceserver.adoc]: I see... The reference was aimed at the two lines. I thought it was just aimed at the second code referenced. My bad!

**Comment by jzheaux** [comment]
@alexandre-melard Are you able to apply the requested changes?

**Comment by alexandre-melard** [comment]
Hi,

I have been too busy this week, I'll look into your comments early next
week.

Cheers

Alex


Le ven. 22 nov. 2019 à 00:43, Josh Cummings <notifications@github.com> a
écrit :

**Comment by jzheaux** [comment]
This is actually obsolete now since #7733 was merged. However, @alexandre-melard, feel free to take a look at the existing docs and see if you'd like to make any more improvements!
